### PR TITLE
Test if *ring* compiles without default features.

### LINKED ISSUE #222: Add `--no-default-features` configurations to CI
We need to verify that people don't break the builds when the functionality that uses the heap is disabled, in particular.

### PR DESCRIPTION
**Pull request by pyfisch**
Closes #222

### CODE REVIEW

File: mk/update-travis-yml.py [hunk 1]
```diff
@@ -61,7 +61,8 @@
 
 modes = [
     "DEBUG",
-    "RELWITHDEBINFO"
+    "RELWITHDEBINFO",
+    "NODEFAULTFEATURES"
```
Comment (djc) [inline mk/update-travis-yml.py]: Nit: add a trailing comma here, so that next time you don't have to change this line if you're adding more.

**Comment by pyfisch** [comment]
fixed nit, added copyright statement to commit message.

**Comment by pyfisch** [comment]
@briansmith running `cargo build --no-default-features` does not assure that the tests pass without `use_heap`.

**Comment by briansmith** [comment]
> @briansmith running cargo build --no-default-features does not assure that the tests pass without use_heap.

Fair enough. I took the "compiles" in the title of your PR too literally.

At any rate, I do think that this is a good thing to do. We just need to find some way to limit the amount of resources this consumes. I suggest that, for example, we limit this to one version of GCC per platform and and Linux only. I don't think the no-heap mode is useful for Mac OS X, iPhone, or Android. That would give us {ARM, Aarch64, x86, x86-64} * {1 version of GCC} * {debug, release} = 8 new configurations to test, instead of 20 new configurations.

The issue is that Travis CI already takes *hours* to finish, because we only get 5 concurrent builds and we have a large number already.

**Pull request closed without merging**